### PR TITLE
libepoxy: fix linking in static targets and add test

### PR DESCRIPTION
**Pull request by starius**
See https://github.com/mxe/mxe/issues/1647#issuecomment-275967915

### CODE REVIEW

File: src/libepoxy.mk [hunk 1]
```diff
@@ -19,7 +19,17 @@ endef
 
 define $(PKG)_BUILD
     cd '$(1)' && autoreconf -fi -I'$(PREFIX)/$(TARGET)/share/aclocal'
-    cd '$(1)' && ./configure \
+    cd '$(1)' && \
+        CFLAGS='$(if $(BUILD_STATIC),-DEPOXY_STATIC,-DEPOXY_SHARED -DEPOXY_DLL)' \
+        ./configure \
         $(MXE_CONFIGURE_OPTS)
     $(MAKE) -C '$(1)' -j '$(JOBS)' install $(MXE_DISABLE_CRUFT)
+    $(SED) 's/Cflags:/Cflags: -DEPOXY_$(if $(BUILD_STATIC),STATIC,SHARED)/' \
```
Comment (tonytheodore) [inline src/libepoxy.mk]: Does the *.pc file also want `-DEPOXY_DLL`

Comment (starius) [inline src/libepoxy.mk]: `-DEPOXY_DLL` means "we are building the DLL now", so it should be set while building the package itself (resulting in `dllexport`) and unset when the package is used by other packages (resulting in `dllimport`).

**Comment by starius** [comment]
Rebased

File: src/libepoxy.mk (hunk 1 above)
Comment (tonytheodore) [inline src/libepoxy.mk]: Thanks for the explanation, it never sticks in my mind.

**Pull request merged**